var.mk

```diff
@@ -14,14 +14,29 @@ PROGRAMATICA_SRC_FILE ?= \
 # The local file gets tried first, and if not usable the remote on gets fetched.
 # If both are unset or set to an empty string, programatica support is skipped.
 
-GHCVERSION := $(shell ghc --numeric-version)
+OSNAME := $(shell uname -s)
+OSVERS := $(shell uname -v 2>/dev/null)
+
+STACK := stack
+STACK_EXEC := $(STACK) exec --
+
+ifeq "$(OSNAME)" "Darwin"
+    STACK_DEPENDENCIES_FLAGS := --flag gtk:have-quartz-gtk
+else
+    STACK_DEPENDENCIES_FLAGS :=
+endif
+
+# Dummy variables to install GHC and install the Haskell-dependencies.  These
+# must be called before ghc or ghc-pkg is called because the packages need to
+# be installed before they are checked.
+STACK_SETUP := $(shell $(STACK) setup > /dev/null)
+STACK_INSTALL_DEPENDENCIES := $(shell $(STACK) build --only-dependencies $(STACK_DEPENDENCIES_FLAGS) > /dev/null)
+
```

Comment:
That's the point where I need the most help. It must be called before all the ghc-pkg calls are made (which are in the var.mk to setup, for instance, `HAXMLVERSION`) 
and I don't know how to do this. As you can see, I am not at all an expert of Make.